src/app/my-messages/deletable-notification/deletable-notification.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { MessagesService } from '../messages.service';
+import { MatDialog } from '@angular/material';
+import { AbstractNotification } from '../abstract-notification';
+import { DeleteConfirmationDialogComponent } from '../../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
```

Comment:
As far as I see the `DeletableNotificationComponent` is used primarily to provide the implementation to delete a notification. It does not have any template and cannot be instantiated independently from other components. So I suggest to remove `@Component` directive from the class, remove `OnInit` interface and make the class abstract. Perhaps it would be good to rename the class into something like `DeletableNotificationComponentTrait`. Here the word `Trait` is used in the same meaning as in Scala language (see https://docs.scala-lang.org/tour/traits.html). It is an interface which can be partially implemented (these are the interfaces with default implementation in Java). What do you think about it?